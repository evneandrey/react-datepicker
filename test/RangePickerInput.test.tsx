import * as React from 'react';
import * as sinon from 'sinon';
import { shallow, mount, ShallowWrapper, ReactWrapper, HTMLAttributes } from 'enzyme';
import RangePickerInput, { FieldType } from '../src/components/RangePickerInput';

describe('<RagePickerInput />', () => {
  let shallowComponent: ShallowWrapper;

  beforeEach(() => {
    shallowComponent = shallow(<RangePickerInput />);
  });

  it('should render without crash', () => {
    expect(shallowComponent).toBeTruthy();
    expect(shallowComponent).toMatchSnapshot();
  });

  describe('event test', () => {
    let onChange: sinon.SinonSpy;
    let onClick: sinon.SinonSpy;
    let mountComponent: ReactWrapper;
    let startInput: ReactWrapper<HTMLAttributes>;
    let endInput: ReactWrapper<HTMLAttributes>;

    beforeEach(() => {
      onChange = sinon.spy();
      onClick = sinon.spy();
      mountComponent = mount(<RangePickerInput onClick={onClick} onChange={onChange} />);
      startInput = mountComponent.find('.range-picker-input__start .picker-input__text');
      endInput = mountComponent.find('.range-picker-input__end .picker-input__text');
    });

    it('should onClick start & end operate separately', () => {
      startInput.simulate('click');
      expect(onClick.getCalls()[0].args[0]).toEqual(FieldType.START);

      endInput.simulate('click');
      expect(onClick.getCalls()[1].args[0]).toEqual(FieldType.END);
      expect(onClick).toHaveProperty('callCount', 2);
    });

    it('should onChange start & end operate separately', () => {
      startInput.simulate('change');
      expect(onChange.getCalls()[0].args[0]).toEqual(FieldType.START);

      endInput.simulate('change');
      expect(onChange.getCalls()[1].args[0]).toEqual(FieldType.END);
      expect(onChange).toHaveProperty('callCount', 2);
    });
  });
});
