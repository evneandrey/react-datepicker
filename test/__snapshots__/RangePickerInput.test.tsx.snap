// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RagePickerInput /> should render without crash 1`] = `
<div
  className="range-picker-input"
>
  <span
    className="range-picker-input__start"
  >
    <PickerInput
      className="range"
      onBlur={[Function]}
      onChange={[Function]}
      onClear={[Function]}
      onClick={[Function]}
    />
  </span>
  <span
    className="range-picker-input__icon"
  >
    <SVGIcon
      color="currentColor"
      id="right-arrow"
      size="16"
    />
  </span>
  <span
    className="range-picker-input__end"
  >
    <PickerInput
      className="range"
      onBlur={[Function]}
      onChange={[Function]}
      onClear={[Function]}
      onClick={[Function]}
    />
  </span>
</div>
`;
